### DISCUSSION TITLE
Access sibling field in an array field from a custom component

### DISCUSSION THREAD

**Discussion by cerize** [payloadcms/payload#1900] [Q&A] [answered]
Hi! Consider I have the following collection Recipes:

```
import { CollectionConfig } from 'payload/types';
import IngredientUnit from '../components/IngredientUnit';

const Recipes: CollectionConfig = {
  slug: 'recipes',
  admin: {
    useAsTitle: 'name',
  },
  fields: [
    {
      name: 'title',
      type: 'text',
    },
    {
        name: 'ingredients',
        type: 'array',
        fields: [
            {
                name: 'name',
                type: 'text'
            },
            {
                name: 'quantity',
                type: 'text'
            },
            {
                name: 'unit',
                type: 'text',
                admin: {
                    components: {
                        Field: IngredientUnit
                    }
                }
            }
        ]
    }
  ],
}

export default Recipes;
```

How can I access the 'name' field inside the custom component 'IngredientUnit'? Using 'useFormFields' I am able to access the 'title' field, but not 'name'.

Thank you!!!

**Comment by JessRynkar** [comment, marked as answer]
Hey @cerize - the useForm() function will allow you to access the sibling data. Not sure how your custom field is setup but it would look something like this:

![Screen Shot 2023-02-09 at 6 05 10 PM](https://user-images.githubusercontent.com/67977755/217899734-ed752024-b426-48e3-880f-172468dcad88.png)

**Reply by SebastianTroc** [reply]
Hi!
any known way to get a setter function to modify the sibling's value? Not only the value like `getSiblingData` allows
